src/main/java/org/oscii/detokenize/TokenLabel.java

```diff
@@ -0,0 +1,79 @@
+package org.oscii.detokenize;
+
+import com.google.gson.Gson;
+import com.ibm.icu.lang.UCharacter;
+
+import java.util.List;
+
+/**
+ * Labels describes how to render a token sequence as a string.
+ */
+public class TokenLabel {
+    boolean capitalize = false;
+    String following = " ";
+    String replace = null;
+
+    public TokenLabel(boolean capitalize, String following, String replace) {
+        this.capitalize = capitalize;
+        this.following = following;
+        this.replace = replace;
+    }
+
+    public String toString() {
+        return new Gson().toJson(this, this.getClass());
+    }
+
+    public static TokenLabel interpret(String json) {
+        TokenLabel label = new Gson().fromJson(json, TokenLabel.class);
+        assert label.following != null;
+        assert label.replace != null;
+        return label;
+    }
+
+    public String renderToken(String token) {
+        if (replace != null) {
+            token = replace;
+        }
+        if (capitalize) {
+            // TODO(denero) Casing should require a locale.
```

Comment:
why? I don't see that capitalization is locale-dependent. Locales are set incorrectly most of the time and the source of really weird errors. If possible, I'd avoid using locales.